src/main/java/org/jpeek/calculus/java/Ccm.java

```diff
@@ -52,57 +64,208 @@ public XML node(
+        final List<XML> allowed = new ArrayList<>(0);
+        for (final XML method : clazz.nodes("methods/method")) {
+            final String name = method.xpath("@name").get(0);
+            if (isConstructorExcluded(params, method) || isStaticMethodExcluded(params, method)) {
+                continue;
+            }
+            allowed.add(method);
+            parents.put(name, name);
+        }
+        for (final XML method : allowed) {
+            final String name = method.xpath("@name").get(0);
+            final List<XML> ops = method.nodes("ops/op");
+            for (final XML operation : ops) {
+                final String code = operation.xpath("@code").get(0);
+                if (code.equals("call")) {
+                    final String classpath = operation.nodes("name").get(0).node().getTextContent();
+                    final List<String> splits = Arrays.asList(classpath.split("\\."));
+                    if (parents.keySet().contains(splits.get(splits.size() - 1))) {
+                        UnionFind.unite(name, splits.get(splits.size() - 1), parents);
+                    }
+                } else {
+                    final String var = operation.node().getTextContent();
+                    if (connections.containsKey(var)) {
+                        connections.get(var).add(name);
+                    } else {
+                        final List<String> init = new ArrayList<>(0);
+                        init.add(name);
+                        connections.put(var, init);
+                    }
+                }
+            }
+        }
+        return UnionFind.runUnionFind(parents, connections);
+    }
+
+    /**
+     * Checks if a static method should be excluded based on parameters.
+     * @param params Parameters for filtering.
+     * @param method The method XML node.
+     * @return True if the method should be excluded, false otherwise.
+     */
+    private static boolean isStaticMethodExcluded(final Map<String, Object> params,
+        final XML method) {
+        return !params.containsKey("include-static-methods") && method.xpath("@static").get(0)
+            .equals("true");
+    }
+
+    /**
+     * Checks if a constructor should be excluded based on parameters.
+     * @param params Parameters for filtering.
+     * @param method The method XML node.
+     * @return True if the constructor should be excluded, false otherwise.
+     */
+    private static boolean isConstructorExcluded(final Map<String, Object> params,
+        final XML method) {
+        return !params.containsKey("include-ctors") && method.xpath("@ctor").get(0).equals("true");
+    }
+
+    /**
+     * Utility class implementing the Union-Find algorithm.
+     * The UnionFind class provides methods to perform the Union-Find algorithm,
+     * which is used to calculate the number of components in a given structure.
+     * @since 0.30.25
+     */
+    private static class UnionFind {
```

Comment:
why it can be normal class, for example GraphClass with constructor with XML with class description and with method to get components list?